src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -141,6 +141,10 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
             {
                 throw new ArgumentException(SR.Argument_NewMMFAppendModeNotAllowed, nameof(mode));
             }
+            if (mode == FileMode.Truncate)
+            {
+                throw new ArgumentException(SR.Argument_NewMMFTruncateModeNotAllowed, nameof(mode));
```

Comment:
I do not have an strong opinion on this one. I might have a small preference to make this one match the netfx behavior:

- We are disallowing `FileMode.Append` already. Is there a good reason for it? If we were to allow `FileMode.Truncate`, should we allow `FileMode.Append` too?
- This API is trying to undo any side-effects (delete the file if it was created) if something fails half-way. The side-effects cannot be undone for `FileMode.Truncate`.